modules/brew/README.md

```diff
@@ -0,0 +1,147 @@
+# brew
+
+
+
+The brew module installs Homebrew (Brew) on your system and ensures the package manager remains updated and maintained. This module sets up systemd services to periodically update and upgrade the installed Brew packages.
+
+
+
+## Features
+
+
+
+- Installs Brew at build time.
+
+- Configures and installs specified Brew packages.
+
+- Sets up systemd services to update and upgrade Brew packages automatically.
+
+- Options to control the frequency of updates and upgrades.
+
+
+
+## Configuration Options
+
+
+
+### `install_brew` (optional: boolean, default: true)
```

Comment:
What is the module used for if not installing brew? How do we expect the services to add brew packages and update brew to work if brew isn't installed?

Is there a possibility of conflict if a user has brew installed prior to switching to an image with brew pre-installed, which would make not pre-installing brew while using the module a valid choice? If so, we should document that very clearly as a warning.

IMO (and with the limited information I have), just including `- type: brew` should be enough to install brew, such as including `- type: yafti` installs yafti.

Reply:
> What is the module used for if not installing brew? How do we expect the services to add brew packages and update brew to work if brew isn't installed?
> 
> Is there a possibility of conflict if a user has brew installed prior to switching to an image with brew pre-installed, which would make not pre-installing brew while using the module a valid choice? If so, we should document that very clearly as a warning.
> 
I drew inspiration from the [chezmoi module](https://github.com/blue-build/modules/blob/main/modules/chezmoi/module.yml#L7), which does have an "install" boolean. 

> IMO (and with the limited information I have), just including `- type: brew` should be enough to install brew, such as including `- type: yafti` installs yafti.
I agree, will remove that option :+1:

Reply:
> I drew inspiration from the chezmoi module, which does have an "install" boolean.

That module has some issues too, which unfortunately made it through review. Trying to catch those before merge in the future.